### DISCUSSION TITLE
Unable to use local piodide with worker

### DISCUSSION THREAD

**Discussion by ninjaclown123** [pyscript/pyscript#2046] [Q&A] [answered]
I would like to use PyScript offline and I followed the steps from "[Use Offline](https://docs.pyscript.net/2024.4.1/user-guide/offline/)" section to get it working. I am able to run the "Hello from PyScript" example perfectly fine. However, my project requires use of the terminal and being able to input into the terminal. For this I need to enable "worker" on PyScript. 

My problem is that when I enable worker, I get the following error:

![image](https://github.com/pyscript/pyscript/assets/135505235/e7303698-f612-4c3f-bd81-1497208b2d53)

![image](https://github.com/pyscript/pyscript/assets/135505235/1e80a713-d52b-4ef0-9b28-1c0727553513)

The error message says 'Failed to resolve module specifier '/pyodide/pyodide.mjs'. However, it works fine when I run the same but without the "worker" keyword. 

Am I doing something wrong or is this a new issue?

Additional: 
TypeError: Failed to resolve module specifier '/pyodide/pyodide.mjs' at Object.get (blob:http://127.0.0.1:5500/4c87cef0-a498-4f9c-b604-29b1d7afd74b:2:27124) at blob:http://127.0.0.1:5500/4c87cef0-a498-4f9c-b604-29b1d7afd74b:2:27955 at async blob:http://127.0.0.1:5500/4c87cef0-a498-4f9c-b604-29b1d7afd74b:2:29192

TypeError: Failed to resolve module specifier '/pyodide/pyodide.mjs'
    at Object.get (blob:http://127.0.0.1:5500/4c87cef0-a498-4f9c-b604-29b1d7afd74b:2:27124)
    at blob:http://127.0.0.1:5500/4c87cef0-a498-4f9c-b604-29b1d7afd74b:2:27955
    at async blob:http://127.0.0.1:5500/4c87cef0-a498-4f9c-b604-29b1d7afd74b:2:29192

I inspected the network tab and found the following:
![image](https://github.com/pyscript/pyscript/assets/135505235/4068111e-a370-4d0a-b843-0108a7b14c52)

I point this out as I noticed that "4c87cef0-a498-4f9c-b604-29b1d7afd74b" also appears in the error message. Could be relevant.

**Comment by WebReflection** [comment, marked as answer]
@ninjaclown123 does it work if you use a fully qualified URL for the interpreter?

**Reply by WebReflection** [reply]
FYI this has been addressed in here https://github.com/pyscript/pyscript/pull/2050

You can try the canary build deployed on *npm* by using `https://cdn.jsdelivr.net/npm/@pyscript/core@0.4.30/dist/core.js` to include *JS* and `https://cdn.jsdelivr.net/npm/@pyscript/core@0.4.30/dist/core.css` to include the *CSS* files, instead of official release channels.

**Reply by ninjaclown123** [reply]
Using a fully qualified URL fixed my issue. Thank you!
